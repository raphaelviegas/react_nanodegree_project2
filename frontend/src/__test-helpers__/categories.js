export const initialState = { categories: [] }

export const categories = [
    {
        name: 'react',
        path: 'react'
    },
    {
        name: 'redux',
        path: 'redux'
    },
    {
        name: 'udacity',
        path: 'udacity'
    }
]