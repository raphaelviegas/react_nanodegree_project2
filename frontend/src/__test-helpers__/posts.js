const uuidv4 = require('uuid/v4')

export const initialState = {
    posts: [],
    sortByOption: '-voteScore',
    invalidPost: false,
    loadingPosts: true,
}

export const posts = [
    {
        id: '8xf0y6ziyjabvozdd253nd',
        timestamp: 1467166872634,
        title: 'Udacity is the best place to learn React',
        body: 'Everyone says so after all.',
        author: 'thingtwo',
        category: 'react',
        voteScore: 10,
        deleted: false,
        commentCount: 2
    },
    {
        id: '6ni6ok3ym7mf1p33lnez',
        timestamp: 1468479767190,
        title: 'Learn Redux in 10 minutes!',
        body: 'Just kidding. It takes more than 10 minutes to learn technology.',
        author: 'thingone',
        category: 'redux',
        voteScore: -5,
        deleted: false,
        commentCount: 0
    }
]

export const removePost = '8xf0y6ziyjabvozdd253nd'

export const votePost = {
    id: '8xf0y6ziyjabvozdd253nd',
    voteScore: 10
}

let id = uuidv4()
export const addPostInfo = {
    author: "Test Author",
    body: "Test Body",
    category: "Test",
    title: "Test Title",
    timestamp: Date.now(),
    id
}

export const editPostInfo = {
    id: '8xf0y6ziyjabvozdd253nd',
    editContent: {
        title: 'Edited post title',
        body: 'Edited post body'
    }
}