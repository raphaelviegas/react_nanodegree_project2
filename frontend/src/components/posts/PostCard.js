import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import MaterialIcon from 'material-icons-react'
import VoteScore from '../shared/VoteScore'
import EditPostForm from '../posts/editPost/EditPostForm'

class PostCard extends Component {
    static propTypes = {
        post: PropTypes.object.isRequired,
        onRemovePost: PropTypes.func.isRequired
    }

    state = {
        showEdit: '',
    }

    showEdit = () => this.setState(() => ({ showEdit: 'card-reveal-active' }))
    hideEdit = () => this.setState(() => ({ showEdit: '' }))

    removePost(id) {
        this.props.onRemovePost(id)
    }
    render() {
        const { id, title, body, author, voteScore, commentCount, category } = this.props.post
        const { showEdit } = this.state
        return (
            <div className="col s12 m6">
                <div className="card grey lighten-4">
                    <div className="card-content">
                        <span className="card-title truncate">{title}</span>
                        <span className="card-author">{author}</span>
                        <p>{body}</p>
                    </div>
                    <div className="card-action valign-wrapper">
                        <VoteScore size={16} score={voteScore} type='post' id={id} />
                        <Link className="align-right" to={`/${category}/${id}`}>Show Post</Link>
                    </div>
                    <div className="card-comments">
                        <span className="card-comments-number">{commentCount}</span>
                    </div>
                    <div className="card-buttons">
                        <div className="card-button-edit" onClick={() => this.showEdit()}>
                            <MaterialIcon icon="edit" size={14} />
                        </div>
                        <div className="card-button-delete" onClick={() => this.removePost(id)}>
                            <MaterialIcon icon="delete" size={14} />
                        </div>
                    </div>
                    <div className={`card-reveal ${showEdit}`}>
                        <EditPostForm
                            id={id}
                            defaultTitle={title}
                            defaultBody={body}
                            onPostCardFormSubmited={this.hideEdit}
                        />
                        <div className="hide-edit" onClick={() => this.hideEdit()}>
                            <MaterialIcon icon="close" color="#c3c3c3" />
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default PostCard
