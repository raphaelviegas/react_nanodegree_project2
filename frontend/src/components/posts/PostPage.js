import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter, Redirect } from 'react-router-dom'
import { fetchPosts, fetchRemovePost } from '../../actions/posts'
import MaterialIcon from 'material-icons-react'
import Moment from 'react-moment';
import VoteScore from '../shared/VoteScore'
import Comments from '../comments/Comments'
import EditPostModal from '../posts/editPost/EditPostModal'

export class PostPage extends Component {
    state = {
        modalOpen: ''
    }

    openModal = () => this.setState(() => ({ modalOpen: 'modal-open' }))
    closeModal = () => this.setState(() => ({ modalOpen: '' }))

    componentDidMount() {
        this.props.fetchPosts()
    }

    componentDidUpdate(prevProps) {
        if (this.props.id !== prevProps.id) {
            this.props.fetchPosts()
        }
    }

    removePost = (id) => {
        this.props.fetchRemovePost(id)
        this.props.history.push('/')
    }

    render() {
        const { postInfo } = this.props
        let invalidPost = (postInfo.length > 0) ? false : true
        return (
            <div>
                {invalidPost
                    ? <Redirect to="/error" />
                    : <div className="post-page">
                        {postInfo.map((post) => (
                            <div key={post.id}>
                                <div className="post-buttons">
                                    <div className="post-button-edit" onClick={() => this.openModal()}>
                                        <MaterialIcon icon="edit" size={20} />
                                    </div>
                                    <div className="post-button-remove" onClick={() => this.removePost(post.id)}>
                                        <MaterialIcon icon="delete" size={20} />
                                    </div>
                                </div>
                                <div className="container">
                                    <h4 className="post-title">{post.title}</h4>
                                    <div className="post-infos">
                                        <span className="post-author">Author: {post.author}</span>
                                        <span className="post-category">Category: {post.category}</span>
                                        <span className="post-date">
                                            <MaterialIcon icon="calendar_today" size={16} />
                                            <Moment format="DD/MMM/YY">{post.timestamp}</Moment>
                                        </span>
                                    </div>
                                    <VoteScore size={18} score={post.voteScore} type='post' id={post.id} />
                                    <p className="post-body">{post.body}</p>
                                    <Comments parentId={post.id} />
                                </div>
                                <EditPostModal
                                    modalOpen={this.state.modalOpen}
                                    onCloseModal={this.closeModal}
                                    id={post.id}
                                    title={post.title}
                                    body={post.body}
                                />
                            </div>
                        ))}
                    </div>
                }
            </div>
        )
    }
}
const mapStateToProps = (state, ownProps) => {
    return {
        id: ownProps.match.params.postId,
        postInfo: state.postsReducer.posts.filter(post => post.id === ownProps.match.params.postId)
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchPosts: () => dispatch(fetchPosts()),
        fetchRemovePost: (id) => dispatch(fetchRemovePost(id))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PostPage));
