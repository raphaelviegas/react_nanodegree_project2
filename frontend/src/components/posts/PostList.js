import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { fetchRemovePost } from '../../actions/posts'
import PostCard from './PostCard'
import sortBy from 'sort-by'

export class PostList extends Component {
  static propTypes = {
    posts: PropTypes.array.isRequired,
    sortByOption: PropTypes.string.isRequired
  }

  removePost = (id) => {
    this.props.fetchRemovePost(id)
  }

  render() {
    const { posts, sortByOption } = this.props
    //SortBy
    const sortedPosts = [...posts].sort(sortBy(sortByOption))
    return (
      <div>
        {posts.length === 0
          ? <div className="noPosts"><h4>No posts to show</h4></div>
          : <div className="row">
            {
              sortedPosts.map((post, index) => {
                return <PostCard post={post} onRemovePost={this.removePost} key={index} />
              })
            }
          </div>
        }
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    posts: ownProps.posts,
    sortByOption: state.postsReducer.sortByOption
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchRemovePost: (id) => dispatch(fetchRemovePost(id))
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PostList));
