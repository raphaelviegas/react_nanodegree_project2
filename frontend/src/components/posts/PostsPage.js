import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { fetchPosts } from '../../actions/posts'
import PostList from '../posts/PostList'
import OrderPosts from '../shared/OrderPosts'

export class PostsPage extends Component {
    static propTypes = {
        posts: PropTypes.array.isRequired
    }

    componentDidMount() {
        this.props.fetchPosts()
        let elems = document.querySelectorAll('.dropdown-trigger');
        // eslint-disable-next-line
        let instances = M.Dropdown.init(elems, { alignment: 'right', constrainWidth: false })
    }

    render() {
        return (
            <div>
                <div className="category-title">
                    Showing All Posts
                <OrderPosts />
                </div>
                <div className="divider"></div>
                <PostList posts={this.props.posts} />
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        posts: state.postsReducer.posts,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchPosts: () => dispatch(fetchPosts())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(PostsPage)
