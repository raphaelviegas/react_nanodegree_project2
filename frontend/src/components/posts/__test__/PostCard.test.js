import React from 'react'
import { shallow } from 'enzyme'
import PostCard from '../PostCard'
import { posts } from '../../../__test-helpers__/posts'

describe('<PostCard />', () => {
    const setup = {
        post: posts[0],
        onRemovePost: jest.fn()
    }
    let wrapper
    beforeEach(() => {
        setup.onRemovePost.mockClear()
        wrapper = shallow(<PostCard {...setup} />)
    })

    it('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Should render a VoteScore component', () => {
        expect(wrapper.find('Connect(VoteScore)')).toHaveLength(1)
    })

    it('Should render a Link component to the post page', () => {
        expect(wrapper.find('Link')).toHaveLength(1)
    })

    it('Should render a EditPostForm component', () => {
        expect(wrapper.find('withRouter(Connect(EditPostForm))')).toHaveLength(1)
    })

    it('Should render initial empty initial state', () => {
        expect(wrapper.state()).toEqual({ showEdit: '' })
    })

    it('Should change state.showEdit to card-reveal-active when method showEdit is called', () => {
        expect(wrapper.state()).toEqual({ showEdit: '' })
        wrapper.instance().showEdit()
        expect(wrapper.state()).toEqual({ showEdit: 'card-reveal-active' })
    })

    it('Should change state.showEdit to empty when method hideEdit is called', () => {
        expect(wrapper.state()).toEqual({ showEdit: '' })
        wrapper.instance().showEdit()
        expect(wrapper.state()).toEqual({ showEdit: 'card-reveal-active' })
        wrapper.instance().hideEdit()
        expect(wrapper.state()).toEqual({ showEdit: '' })
    })

    it('Should call onRemovePost when method removePost is called', () => {
        expect(setup.onRemovePost).toHaveBeenCalledTimes(0)
        wrapper.instance().removePost()
        expect(setup.onRemovePost).toHaveBeenCalledTimes(1)
    })

    it('Should call method showEdit when .card-button-edit is clicked', () => {
        const mockshowEdit = jest.fn()
        wrapper.instance().showEdit = mockshowEdit
        expect(mockshowEdit).toHaveBeenCalledTimes(0)
        wrapper.find('.card-button-edit').simulate('click')
        expect(mockshowEdit).toHaveBeenCalledTimes(1)
    })

    it('Should call method removePost when .card-button-delete is clicked', () => {
        const mockremovePost = jest.fn()
        wrapper.instance().removePost = mockremovePost
        expect(mockremovePost).toHaveBeenCalledTimes(0)
        wrapper.find('.card-button-delete').simulate('click')
        expect(mockremovePost).toHaveBeenCalledTimes(1)
    })

    it('Should call method hideEdit when .hide-edit is clicked', () => {
        const mockhideEdit = jest.fn()
        wrapper.instance().hideEdit = mockhideEdit
        expect(mockhideEdit).toHaveBeenCalledTimes(0)
        wrapper.find('.hide-edit').simulate('click')
        expect(mockhideEdit).toHaveBeenCalledTimes(1)
    })
})