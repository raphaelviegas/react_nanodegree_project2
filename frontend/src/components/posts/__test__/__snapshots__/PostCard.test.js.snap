// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostCard
    onRemovePost={[Function]}
    post={
        Object {
            "author": "thingtwo",
            "body": "Everyone says so after all.",
            "category": "react",
            "commentCount": 2,
            "deleted": false,
            "id": "8xf0y6ziyjabvozdd253nd",
            "timestamp": 1467166872634,
            "title": "Udacity is the best place to learn React",
            "voteScore": 10,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card grey lighten-4"
>
        <div
                className="card-content"
        >
                <span
                        className="card-title truncate"
                >
                        Udacity is the best place to learn React
                </span>
                <span
                        className="card-author"
                >
                        thingtwo
                </span>
                <p>
                        Everyone says so after all.
                </p>
        </div>
        <div
                className="card-action valign-wrapper"
        >
                <Connect(VoteScore)
                        id="8xf0y6ziyjabvozdd253nd"
                        score={10}
                        size={16}
                        type="post"
                />
                <Link
                        className="align-right"
                        replace={false}
                        to="/react/8xf0y6ziyjabvozdd253nd"
                >
                        Show Post
                </Link>
        </div>
        <div
                className="card-comments"
        >
                <span
                        className="card-comments-number"
                >
                        2
                </span>
        </div>
        <div
                className="card-buttons"
        >
                <div
                        className="card-button-edit"
                        onClick={[Function]}
                >
                        <MaterialIcon
                                icon="edit"
                                size={14}
                        />
                </div>
                <div
                        className="card-button-delete"
                        onClick={[Function]}
                >
                        <MaterialIcon
                                icon="delete"
                                size={14}
                        />
                </div>
        </div>
        <div
                className="card-reveal "
        >
                <withRouter(Connect(EditPostForm))
                        defaultBody="Everyone says so after all."
                        defaultTitle="Udacity is the best place to learn React"
                        id="8xf0y6ziyjabvozdd253nd"
                        onPostCardFormSubmited={[Function]}
                />
                <div
                        className="hide-edit"
                        onClick={[Function]}
                >
                        <MaterialIcon
                                color="#c3c3c3"
                                icon="close"
                        />
                </div>
        </div>
</div>,
      "className": "col s12 m6",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-content"
>
            <span
                        className="card-title truncate"
            >
                        Udacity is the best place to learn React
            </span>
            <span
                        className="card-author"
            >
                        thingtwo
            </span>
            <p>
                        Everyone says so after all.
            </p>
</div>,
          <div
            className="card-action valign-wrapper"
>
            <Connect(VoteScore)
                        id="8xf0y6ziyjabvozdd253nd"
                        score={10}
                        size={16}
                        type="post"
            />
            <Link
                        className="align-right"
                        replace={false}
                        to="/react/8xf0y6ziyjabvozdd253nd"
            >
                        Show Post
            </Link>
</div>,
          <div
            className="card-comments"
>
            <span
                        className="card-comments-number"
            >
                        2
            </span>
</div>,
          <div
            className="card-buttons"
>
            <div
                        className="card-button-edit"
                        onClick={[Function]}
            >
                        <MaterialIcon
                                    icon="edit"
                                    size={14}
                        />
            </div>
            <div
                        className="card-button-delete"
                        onClick={[Function]}
            >
                        <MaterialIcon
                                    icon="delete"
                                    size={14}
                        />
            </div>
</div>,
          <div
            className="card-reveal "
>
            <withRouter(Connect(EditPostForm))
                        defaultBody="Everyone says so after all."
                        defaultTitle="Udacity is the best place to learn React"
                        id="8xf0y6ziyjabvozdd253nd"
                        onPostCardFormSubmited={[Function]}
            />
            <div
                        className="hide-edit"
                        onClick={[Function]}
            >
                        <MaterialIcon
                                    color="#c3c3c3"
                                    icon="close"
                        />
            </div>
</div>,
        ],
        "className": "card grey lighten-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-title truncate"
>
                Udacity is the best place to learn React
</span>,
              <span
                className="card-author"
>
                thingtwo
</span>,
              <p>
                Everyone says so after all.
</p>,
            ],
            "className": "card-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Udacity is the best place to learn React",
                "className": "card-title truncate",
              },
              "ref": null,
              "rendered": "Udacity is the best place to learn React",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "thingtwo",
                "className": "card-author",
              },
              "ref": null,
              "rendered": "thingtwo",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Everyone says so after all.",
              },
              "ref": null,
              "rendered": "Everyone says so after all.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(VoteScore)
                id="8xf0y6ziyjabvozdd253nd"
                score={10}
                size={16}
                type="post"
/>,
              <Link
                className="align-right"
                replace={false}
                to="/react/8xf0y6ziyjabvozdd253nd"
>
                Show Post
</Link>,
            ],
            "className": "card-action valign-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "8xf0y6ziyjabvozdd253nd",
                "score": 10,
                "size": 16,
                "type": "post",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Show Post",
                "className": "align-right",
                "replace": false,
                "to": "/react/8xf0y6ziyjabvozdd253nd",
              },
              "ref": null,
              "rendered": "Show Post",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="card-comments-number"
>
              2
</span>,
            "className": "card-comments",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "card-comments-number",
            },
            "ref": null,
            "rendered": 2,
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-button-edit"
                onClick={[Function]}
>
                <MaterialIcon
                                icon="edit"
                                size={14}
                />
</div>,
              <div
                className="card-button-delete"
                onClick={[Function]}
>
                <MaterialIcon
                                icon="delete"
                                size={14}
                />
</div>,
            ],
            "className": "card-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MaterialIcon
                  icon="edit"
                  size={14}
/>,
                "className": "card-button-edit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": "edit",
                  "size": 14,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MaterialIcon
                  icon="delete"
                  size={14}
/>,
                "className": "card-button-delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": "delete",
                  "size": 14,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(Connect(EditPostForm))
                defaultBody="Everyone says so after all."
                defaultTitle="Udacity is the best place to learn React"
                id="8xf0y6ziyjabvozdd253nd"
                onPostCardFormSubmited={[Function]}
/>,
              <div
                className="hide-edit"
                onClick={[Function]}
>
                <MaterialIcon
                                color="#c3c3c3"
                                icon="close"
                />
</div>,
            ],
            "className": "card-reveal ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "defaultBody": "Everyone says so after all.",
                "defaultTitle": "Udacity is the best place to learn React",
                "id": "8xf0y6ziyjabvozdd253nd",
                "onPostCardFormSubmited": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MaterialIcon
                  color="#c3c3c3"
                  icon="close"
/>,
                "className": "hide-edit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "#c3c3c3",
                  "icon": "close",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card grey lighten-4"
>
          <div
                    className="card-content"
          >
                    <span
                              className="card-title truncate"
                    >
                              Udacity is the best place to learn React
                    </span>
                    <span
                              className="card-author"
                    >
                              thingtwo
                    </span>
                    <p>
                              Everyone says so after all.
                    </p>
          </div>
          <div
                    className="card-action valign-wrapper"
          >
                    <Connect(VoteScore)
                              id="8xf0y6ziyjabvozdd253nd"
                              score={10}
                              size={16}
                              type="post"
                    />
                    <Link
                              className="align-right"
                              replace={false}
                              to="/react/8xf0y6ziyjabvozdd253nd"
                    >
                              Show Post
                    </Link>
          </div>
          <div
                    className="card-comments"
          >
                    <span
                              className="card-comments-number"
                    >
                              2
                    </span>
          </div>
          <div
                    className="card-buttons"
          >
                    <div
                              className="card-button-edit"
                              onClick={[Function]}
                    >
                              <MaterialIcon
                                        icon="edit"
                                        size={14}
                              />
                    </div>
                    <div
                              className="card-button-delete"
                              onClick={[Function]}
                    >
                              <MaterialIcon
                                        icon="delete"
                                        size={14}
                              />
                    </div>
          </div>
          <div
                    className="card-reveal "
          >
                    <withRouter(Connect(EditPostForm))
                              defaultBody="Everyone says so after all."
                              defaultTitle="Udacity is the best place to learn React"
                              id="8xf0y6ziyjabvozdd253nd"
                              onPostCardFormSubmited={[Function]}
                    />
                    <div
                              className="hide-edit"
                              onClick={[Function]}
                    >
                              <MaterialIcon
                                        color="#c3c3c3"
                                        icon="close"
                              />
                    </div>
          </div>
</div>,
        "className": "col s12 m6",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-content"
>
              <span
                            className="card-title truncate"
              >
                            Udacity is the best place to learn React
              </span>
              <span
                            className="card-author"
              >
                            thingtwo
              </span>
              <p>
                            Everyone says so after all.
              </p>
</div>,
            <div
              className="card-action valign-wrapper"
>
              <Connect(VoteScore)
                            id="8xf0y6ziyjabvozdd253nd"
                            score={10}
                            size={16}
                            type="post"
              />
              <Link
                            className="align-right"
                            replace={false}
                            to="/react/8xf0y6ziyjabvozdd253nd"
              >
                            Show Post
              </Link>
</div>,
            <div
              className="card-comments"
>
              <span
                            className="card-comments-number"
              >
                            2
              </span>
</div>,
            <div
              className="card-buttons"
>
              <div
                            className="card-button-edit"
                            onClick={[Function]}
              >
                            <MaterialIcon
                                          icon="edit"
                                          size={14}
                            />
              </div>
              <div
                            className="card-button-delete"
                            onClick={[Function]}
              >
                            <MaterialIcon
                                          icon="delete"
                                          size={14}
                            />
              </div>
</div>,
            <div
              className="card-reveal "
>
              <withRouter(Connect(EditPostForm))
                            defaultBody="Everyone says so after all."
                            defaultTitle="Udacity is the best place to learn React"
                            id="8xf0y6ziyjabvozdd253nd"
                            onPostCardFormSubmited={[Function]}
              />
              <div
                            className="hide-edit"
                            onClick={[Function]}
              >
                            <MaterialIcon
                                          color="#c3c3c3"
                                          icon="close"
                            />
              </div>
</div>,
          ],
          "className": "card grey lighten-4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="card-title truncate"
>
                  Udacity is the best place to learn React
</span>,
                <span
                  className="card-author"
>
                  thingtwo
</span>,
                <p>
                  Everyone says so after all.
</p>,
              ],
              "className": "card-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Udacity is the best place to learn React",
                  "className": "card-title truncate",
                },
                "ref": null,
                "rendered": "Udacity is the best place to learn React",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "thingtwo",
                  "className": "card-author",
                },
                "ref": null,
                "rendered": "thingtwo",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Everyone says so after all.",
                },
                "ref": null,
                "rendered": "Everyone says so after all.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(VoteScore)
                  id="8xf0y6ziyjabvozdd253nd"
                  score={10}
                  size={16}
                  type="post"
/>,
                <Link
                  className="align-right"
                  replace={false}
                  to="/react/8xf0y6ziyjabvozdd253nd"
>
                  Show Post
</Link>,
              ],
              "className": "card-action valign-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "8xf0y6ziyjabvozdd253nd",
                  "score": 10,
                  "size": 16,
                  "type": "post",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Show Post",
                  "className": "align-right",
                  "replace": false,
                  "to": "/react/8xf0y6ziyjabvozdd253nd",
                },
                "ref": null,
                "rendered": "Show Post",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="card-comments-number"
>
                2
</span>,
              "className": "card-comments",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "className": "card-comments-number",
              },
              "ref": null,
              "rendered": 2,
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-button-edit"
                  onClick={[Function]}
>
                  <MaterialIcon
                                    icon="edit"
                                    size={14}
                  />
</div>,
                <div
                  className="card-button-delete"
                  onClick={[Function]}
>
                  <MaterialIcon
                                    icon="delete"
                                    size={14}
                  />
</div>,
              ],
              "className": "card-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <MaterialIcon
                    icon="edit"
                    size={14}
/>,
                  "className": "card-button-edit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "edit",
                    "size": 14,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <MaterialIcon
                    icon="delete"
                    size={14}
/>,
                  "className": "card-button-delete",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "delete",
                    "size": 14,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <withRouter(Connect(EditPostForm))
                  defaultBody="Everyone says so after all."
                  defaultTitle="Udacity is the best place to learn React"
                  id="8xf0y6ziyjabvozdd253nd"
                  onPostCardFormSubmited={[Function]}
/>,
                <div
                  className="hide-edit"
                  onClick={[Function]}
>
                  <MaterialIcon
                                    color="#c3c3c3"
                                    icon="close"
                  />
</div>,
              ],
              "className": "card-reveal ",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "defaultBody": "Everyone says so after all.",
                  "defaultTitle": "Udacity is the best place to learn React",
                  "id": "8xf0y6ziyjabvozdd253nd",
                  "onPostCardFormSubmited": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <MaterialIcon
                    color="#c3c3c3"
                    icon="close"
/>,
                  "className": "hide-edit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "#c3c3c3",
                    "icon": "close",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
